src/config.cc

```diff
@@ -5,200 +5,198 @@
-				}
-				const auto [is_valid, parsed] = parse_uint(value.c_str());
-				if(is_valid) { m_graph_print_max_verts = parsed; }
-			}
-		}
+			const auto has_gpmv = parsed_and_validated_envs.get(env_gpmv);
+			if(has_gpmv) { m_graph_print_max_verts = *has_gpmv; }
 
-		// --------------------------------- CELERITY_DEVICES ---------------------------------
-
-		{
-			const auto [is_set, value] = get_env("CELERITY_DEVICES");
-			if(is_set) {
-				if(value.empty()) {
-					CELERITY_WARN("CELERITY_DEVICES is set but empty - ignoring");
-				} else {
-					std::istringstream ss{value};
-					std::vector<std::string> values{std::istream_iterator<std::string>{ss}, std::istream_iterator<std::string>{}};
-					if(static_cast<long>(m_host_cfg.local_rank) > static_cast<long>(values.size()) - 2) {
-						throw std::runtime_error(fmt::format("Process has local rank {}, but CELERITY_DEVICES only includes {} device(s)",
-						    m_host_cfg.local_rank, values.empty() ? 0 : values.size() - 1));
-					}
-
-					if(static_cast<long>(values.size()) - 1 > static_cast<long>(m_host_cfg.node_count)) {
-						CELERITY_WARN("CELERITY_DEVICES contains {} device indices, but only {} worker processes were spawned on this host", values.size() - 1,
-						    m_host_cfg.node_count);
-					}
-
-					const auto pid_parsed = parse_uint(values[0].c_str());
-					const auto did_parsed = parse_uint(values[m_host_cfg.local_rank + 1].c_str());
-					if(!pid_parsed.first || !did_parsed.first) {
-						CELERITY_WARN("CELERITY_DEVICES contains invalid value(s) - will be ignored");
-					} else {
-						m_device_cfg = device_config{pid_parsed.second, did_parsed.second};
-					}
-				}
-			}
-		}
+			// --------------------------------- CELERITY_DEVICES ---------------------------------
 
-		// ----------------------------- CELERITY_PROFILE_KERNEL ------------------------------
-		{
-			const auto result = get_env("CELERITY_PROFILE_OCL");
-			if(result.first) {
-				CELERITY_WARN("CELERITY_PROFILE_OCL has been renamed to CELERITY_PROFILE_KERNEL with Celerity 0.3.0.");
-				m_enable_device_profiling = result.second == "1";
+			const auto has_devs = parsed_and_validated_envs.get(env_devs);
+			if(has_devs) {
+				m_devices_list = *has_devs;
+				const auto pid_parsed = (*has_devs)[0];
+				const auto did_parsed = (*has_devs)[m_host_cfg.local_rank + 1];
+				m_device_cfg = device_config{pid_parsed, did_parsed};
 			}
-		}
 
-		{
-			const auto result = get_env("CELERITY_PROFILE_KERNEL");
-			if(result.first) { m_enable_device_profiling = result.second == "1"; }
-		}
+			// ----------------------------- CELERITY_PROFILE_KERNEL ------------------------------
 
-		// -------------------------------- CELERITY_FORCE_WG ---------------------------------
+			const auto has_profile_kernel = parsed_and_validated_envs.get(env_profile_kernel);
+			if(has_profile_kernel) { m_enable_device_profiling = *has_profile_kernel; }
 
-		{
-			const auto result = get_env("CELERITY_FORCE_WG");
```

Comment:
`CELERITY_FORCE_WG` is now completely missing, but it should still print the warning that support for it has been dropped if it is specified by the user.